crates/aptos-dkg/tests/sigma_protocol.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_dkg::{
+    algebra::{
+        homomorphism,
+        homomorphism::{DiagonalProductMap, Map},
+    },
+    sigma_protocol,
+};
+use ark_bls12_381::Bls12_381;
+use ark_bn254::Bn254;
+use ark_ec::{pairing::Pairing, CurveGroup, PrimeGroup};
+use ark_ff::UniformRand;
+use ark_serialize::CanonicalSerialize;
+use ark_std::rand::{thread_rng, CryptoRng, RngCore};
+
+#[cfg(test)]
+pub fn test_sigma_protocol<E, P>(instance: P, witness: P::Witness)
+where
+    E: Pairing,
+    P: sigma_protocol::Trait<E>,
+{
+    let mut rng = thread_rng();
+
+    let hom = instance.homomorphism();
+    let statement = hom.apply(&witness);
```

Comment:
a bit odd that you are calling this the statement... applying the homomorphism to the witness would yield the prover's first message, no?

for now, not sure what's happening here; will look above.